### PR TITLE
Feature/isdb 351

### PR DESCRIPTION
**Pull request by adetolacog**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/cafcass/PostcodeLookupService.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            if (StringUtils.isEmpty(key)) {
+                throw new PostcodeValidationException("Postcode API Key is null");
+            }
+            UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url + "/postcode");
+            for (Map.Entry<String, String> entry : params.entrySet()) {
+                builder.queryParam(entry.getKey(), entry.getValue());
+            }
+
+            MultiValueMap<String, String> headers = new HttpHeaders();
+            headers.set("Accept", "application/json");
```
Comment (parvezhmcts) [inline src/main/java/uk/gov/hmcts/reform/prl/services/cafcass/PostcodeLookupService.java:72]: you can use HttpHeaders.ACCEPT and MediaType.APPLICATION_JSON_VALUE.

File: src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java [hunk 2]
```diff
@@ -107,8 +107,8 @@ public class PrlAppsConstants {
     public static final String COURT_NAME = "courtName";
 
     public static final List<String> ROLES = List.of("caseworker-privatelaw-courtadmin",
-                                                     "caseworker-privatelaw-judge",
-                                                     "caseworker-privatelaw-la");
+            "caseworker-privatelaw-judge",
+            "caseworker-privatelaw-la");
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/prl/constants/PrlAppsConstants.java:111]: pls revert this file from release branch

File: src/integrationTest/resources/application.yaml [hunk 3]
```diff
@@ -319,3 +319,8 @@ document:
 southampton:
   court:
     email-address: ${SOUTHAMPTON_COURT_EMAIL_ADDRESS:}
+
+postcodelookup:
+  api:
+    url: https://api.os.uk/search/places/v1
+    key: ${POST_CODE_LOOKUP_KEY:REPLACE_ME}
```
Comment (hmcts-kodatala) [inline src/integrationTest/resources/application.yaml:326]: not required and revert this

File: src/functionalTest/resources/application.yaml [hunk 4]
```diff
@@ -307,3 +307,8 @@ document:
 southampton:
   court:
     email-address: ${SOUTHAMPTON_COURT_EMAIL_ADDRESS:}
+
+postcodelookup:
+  api:
```
Comment (hmcts-kodatala) [inline src/functionalTest/resources/application.yaml:312]: not required and revert this

File: src/contractTest/resources/application.yaml [hunk 5]
```diff
@@ -289,3 +289,8 @@ document:
 southampton:
   court:
     email-address: ${SOUTHAMPTON_COURT_EMAIL_ADDRESS:}
+
+postcodelookup:
+  api:
```
Comment (hmcts-kodatala) [inline src/contractTest/resources/application.yaml:294]: not required and revert this

**Review by hmcts-kodatala — COMMENTED** [review]
pls  resolve all review comments and add post code key in value.yml

File: src/functionalTest/resources/application.yaml (hunk 4 above)
Comment (adetolacog) [reply src/functionalTest/resources/application.yaml:312]: Retaining this until we could find an alternative approach.

**Pull request merged**